internal/commands/app/create.go

```diff
@@ -157,17 +159,13 @@ func (cmd *CommandCreate) Handler(profile *user.Profile, ui terminal.UI, clients
 		return err
 	}
 
-	if err := cmd.inputs.resolveTemplateID(ui, clients.Realm); err != nil {
+	if err = cmd.inputs.resolveTemplateID(ui, clients.Realm); err != nil {
 		return err
 	}
 
-	var dsClusters []dataSourceCluster
-	var dsClustersMissing []string
-	if len(cmd.inputs.Clusters) > 0 {
-		dsClusters, dsClustersMissing, err = cmd.inputs.resolveClusters(ui, clients.Atlas, groupID)
-		if err != nil {
-			return err
-		}
+	dsClusters, dsClustersMissing, err := cmd.inputs.resolveClusters(ui, clients.Atlas, groupID)
```

Comment:
moved the `if len(cmd.inputs.Clusters) > 0` to inside of `resolveClusters`

it will be applied if `cmd.inputs.Template` is an empty string (aka default app creation)